examples/notification-demo/echo_server/echo.c

```diff
@@ -0,0 +1,1975 @@
+
+    /* open file (read-only binary) */
+    lFile = fopen(fname, "rb");
+    if (!lFile) {
+        fprintf(stderr, "Error loading %s\n", fname);
+        return -1;
+    }
+
+    fseek(lFile, 0, SEEK_END);
+    fileSz = (int)ftell(lFile);
+    rewind(lFile);
+    if (fileSz  > 0) {
+        *bufLen = (size_t)fileSz;
+        *buf = (char*)malloc(*bufLen);
+        if (*buf == NULL) {
+            ret = MEMORY_E;
+            fprintf(stderr,
+                    "Error allocating %lu bytes\n", (unsigned long)*bufLen);
+        }
+        else {
+            size_t readLen = fread(*buf, *bufLen, 1, lFile);
+
+            /* check response code */
+            ret = (readLen > 0) ? 0 : -1;
+        }
+    }
+    else {
+        ret = -1;
+    }
+    fclose(lFile);
+
+    return ret;
+}
+#else
+    /* for wolfsentry_config_data */
+    #include "../notify-config.h"
+#endif
+
+int main(int argc, char **argv) {
+    wolfsentry_errcode_t ret;
+    size_t circlog_size;
+    char *http_buf = NULL;
+    size_t http_buf_size = 0, http_buf_len;
+    static const struct io_ctx io_ctx = { 1000 }; /* timeout = 1 second */
+    int i;
+    const char *wolfsentry_configfile = "../notify-config.json";
+    uint64_t circlog_size_uint64;
+#ifndef NO_FILESYSTEM
+    char* wolfsentry_config_data = NULL;
+    size_t wolfsentry_config_data_sz = 0;
+#endif
+
+    http_buf_size = HTTP_BUF_SZ;
+    http_buf = (char *)malloc(http_buf_size);
+    if (http_buf == NULL) {
+        perror("malloc");
+        exit(1);
+    }
+
+#ifdef DEBUG_TLS
+    printf("Echo SSL init\n");
+#endif
+    if (echo_ssl_init() < 0)
+        exit(1);
+
+#ifdef DEBUG_HTTP
+    printf("Sentry init\n");
+#endif
+
+    ret = wolfsentry_user_source_string_set(WOLFSENTRY_SOURCE_ID, __FILE__);
+    if (ret < 0) {
+        fprintf(stderr, "wolfsentry_user_source_string_set() failed: " WOLFSENTRY_ERROR_FMT "\n", WOLFSENTRY_ERROR_FMT_ARGS(ret));
+        exit(1);
+    }
+
+    /* load configuration */
+    for (i=1; i<argc; i+=2) {
+        if (argc < i+2) {
+            fprintf(stderr,"%s: missing argument to \"%s\"\n", argv[0], argv[i]);
+            exit(1);
+        }
+        if (! strcmp(argv[i],"--config")) {
+            wolfsentry_configfile = argv[i+1];
+        }
+        else if (! strcmp(argv[i],"--kv-string")) {
+        }
+        else if (! strcmp(argv[i],"--kv-int")) {
+        }
+        else if (! strcmp(argv[i],"--greenlist")) {
+        }
+        else if (! strcmp(argv[i],"--redlist")) {
+        }
+        else {
+            fprintf(stderr,"%s: unrecognized argument \"%s\"\n",argv[0],argv[i]);
+            exit(1);
+        }
+    }
+
+#ifndef NO_FILESYSTEM
+    printf("Loading configuration file %s\n", wolfsentry_configfile);
+    if (load_file(wolfsentry_configfile, &wolfsentry_config_data, &wolfsentry_config_data_sz) != 0) {
+        fprintf(stderr, "Error loading configuration file %s\n", wolfsentry_configfile);
+        exit(1);
+    }
+    (void)wolfsentry_config_data_sz;
```

Comment:
instead of `(void)`ing this, we should use it, by adding a length arg to `sentry_init()`, which currently is `strlen()`ing the string to pass it to `wolfsentry_config_json_oneshot()`.

Reply:
Can of worms. Not changing